src/network/terrier_server.cpp

```diff
@@ -38,44 +43,113 @@ TerrierServer::TerrierServer(common::ManagedPointer<ProtocolInterpreter::Provide
   signal(SIGPIPE, SIG_IGN);
 }
 
-void TerrierServer::RunServer() {
-  // This line is critical to performance for some reason
-  evthread_use_pthreads();
+template<TerrierServer::SocketType type>
+void TerrierServer::RegisterSocket() {
+  static_assert(type == NETWORKED_SOCKET || type == UNIX_DOMAIN_SOCKET, "There should only be two socket types.");
+
+  constexpr auto conn_backlog = common::Settings::CONNECTION_BACKLOG;
+  constexpr auto is_networked_socket = type == NETWORKED_SOCKET;
+  constexpr std::string_view socket_description = is_networked_socket ? "networked" : "Unix domain";
+
+  auto &socket_fd = is_networked_socket ? network_socket_fd_ : unix_domain_socket_fd_;
+
+  // Gets the appropriate sockaddr for the given SocketType.
+  auto socket_addr = ([&] {
+    if constexpr (is_networked_socket) {
+      struct sockaddr_in sin = {0};
+
+      sin.sin_family = AF_INET;
+      sin.sin_addr.s_addr = INADDR_ANY;
+      sin.sin_port = htons(port_);
 
-  int conn_backlog = common::Settings::CONNECTION_BACKLOG;
+      return sin;
+    } else {
+      // Builds the socket path name
+      // Naming conventions come from https://www.postgresql.org/docs/9.3/runtime-config-connection.html
+      const std::string socket_path = fmt::format("{0}/.s.PGSQL.{1}", socket_directory_, port_);
+      struct sockaddr_un sun = {0};
 
-  struct sockaddr_in sin;
-  std::memset(&sin, 0, sizeof(sin));
-  sin.sin_family = AF_INET;
-  sin.sin_addr.s_addr = INADDR_ANY;
-  sin.sin_port = htons(port_);
+      // Validate pathname
+      if (socket_path.length() >= 108 /* Max Unix socket path length */ ) {
```

Comment:
Allegedly this limit is 104 on Mac OS (which I don't use so I can't confirm): https://serverfault.com/questions/641347/check-if-a-path-exceeds-maximum-for-unix-domain-socket

Would it be appropriate to use `sizeof(sun.sun_path)` instead?

Reply:
Good idea. I think so. I'll do that now.